picoquicfirst/picoquicdemo.c

```diff
@@ -248,6 +248,11 @@ static void first_server_callback(picoquic_cnx_t* cnx,
     printf("Server CB, Stream: %" PRIu64 ", %" PRIst " bytes, fin=%d (%s)\n",
         stream_id, length, fin_or_event, picoquic_log_fin_or_event_name(fin_or_event));
 
+    if (fin_or_event == picoquic_callback_prepare_to_send) {
```

Comment:
Seems like a good candidate for a switch statement now. :)

Reply:
Maybe, but I tried ans it is not obvious. There are several steps in the process, e.g. responses that do need to identify a stream, then responses that work without bothering about stream state, then responses that depend on the stream state. I would rather leave that alone until we actually rewrite the server process, maybe to use the callbacks. Also, we will need to revisit that if we start supporting multiple application protocols, e.g. Http 0.9, Http 1.0, and H3.